packages/@ember/routing/route.ts

```diff
@@ -145,12 +145,14 @@ interface Route<T = unknown> extends IRoute<T> {
     export default class LoginRoute extends Route {
       @action
       didTransition() {
-        this.controller.get('errors.base').clear();
+        // your code there
         return true; // Bubble the didTransition event
       }
     }
     ```
 
+    Example above is deprecated in favor of https://deprecations.emberjs.com/v3.x/#toc_deprecate-router-events
+
```

Comment:
Actually, the `Route.didTransition` and `Route.willTransition` actions were *not* removed: these are per-route lifecycle hooks, while the deprecation was for `Router.didTransition` and `Router.willTransition` (note: `Router`, not `RouterService`). This is easy to get confused by: I myself had to go dig back through all the details! And honestly we probably *should* remove these in favor of "please just use the router service", but I expect we will do so via general router cleanup. Accordingly, let's refer this change:

```suggestion
```